### ISSUE TITLE
Add CloudEvents integration

### ISSUE THREAD

**Issue by sagikazarmark** [ThreeDotsLabs/watermill#125]
[CloudEvents](https://cloudevents.io/) is a specification for sending events in the cloud native environment. They also provide SDKs.

It might be a good addition to the project to support CloudEvents. Although it's not stable yet, as of 0.3 the "core" is considered to be stable enough.

**Comment by roblaszczak** [comment]
We will look on that :) Do you think that may be integrated as a component? Or closer to Pub/Sub?

**Comment by sagikazarmark** [comment]
I think it's closer to Pub/Sub, but I have to admit, I'm not that familiar with the spec. Maybe looking at the implemented SDKs could answer that question.

**Comment by OneCricketeer** [comment]
I feel as though watermill should focus on handling byte slices. Serialization can be handled as middleware layers, or in the processing logic

**Comment by eolalde** [comment]
Indeed, I love watermill and use it in a few projects, and now starting to look into cloudevents.io it seems like a great fit for the watermill landscape.
